# HG changeset patch
# User Rik <rik@octave.org>
# Date 1542328942 28800
#      Thu Nov 15 16:42:22 2018 -0800
# Branch stable
# Node ID 8eae32b6bce078b75c5b8ff13c34530b1e3aa654
# Parent  658153e1976d00b94a2cd81c9b8600177ccd3ee8
Don't segfault for unmatched brace in PATH (bug #55013).

* kpse.cc (brace_gobbler): Check whether routine has found a brace, or just the
end-of-string.  For the latter, return 0 because search failed.

diff --git a/liboctave/util/kpse.cc b/liboctave/util/kpse.cc
--- a/liboctave/util/kpse.cc
+++ b/liboctave/util/kpse.cc
@@ -1115,16 +1115,17 @@ brace_gobbler (const std::string& text, 
 
       if (c == '{')
         level++;
       else if (c == '}' && level)
         level--;
     }
 
   indx = i;
+  c = (c == satisfy) ? c : 0;
   return c;
 }
 
 /* Expand extra colons.  */
 
 /* Check for leading colon first, then trailing, then doubled, since
    that is fastest.  Usually it will be leading or trailing.  */
 
